### PR TITLE
added support for listing the edgegateways of a vdc

### PR DESCRIPTION
**Pull request by floriankammermann**
This change is [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/161)

### CODE REVIEW

File: pyvcloud/vcd/vdc.py [hunk 1]
```diff
@@ -340,6 +342,21 @@ def list_resources(self, entity_type=None):
                     result.append({'name': vapp.get('name')})
         return result
 
+    def list_edgegateways(self):
```
Comment (rocknes) [inline pyvcloud/vcd/vdc.py]: nit : list_edge_gateways

**Comment by pacogomez** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/pyvcloud/161)*

**Comment by pacogomez** [comment]
please see my comments below

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/pyvcloud/161#-:-L1TwxZ0x8GV17-mo35Y:b-vtjpvp)*

File: pyvcloud/vcd/vdc.py (hunk 1 above)
Comment (floriankammermann) [reply pyvcloud/vcd/vdc.py]: method naming is fixed in latest commit fixed in latest commit 
get the edge gateway list on the link is fixed in the latest commit

I had a problem to find the Link `rel="fullEdgeGateway"`, couldn't see it on my test vdc and I also couldn't find it in the docs https://pubs.vmware.com/vcd-820/index.jsp?topic=%2Fcom.vmware.vcloud.api.sp.doc_27_0%2FGUID-EECCA924-4DCE-490F-B352-5C212AD22411.html&resultof=%22%65%64%67%65%47%61%74%65%77%61%79%73%22%20%22%65%64%67%65%67%61%74%65%77%61%69%22%20%22%4c%69%6e%6b%22%20%22%6c%69%6e%6b%22%20

**Comment by pacogomez** [comment]
thanks for the update, please see my comments

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[pyvcloud/vcd/vdc.py, line 345 at r1](https://reviewable.io:443/reviews/vmware/pyvcloud/161#-L1RNOiTSLa6z8BN3MOZ-r1-345:-L1tGZ6wcOWVg0uignna:btgtkhg) ([raw file](https://github.com/vmware/pyvcloud/blob/89bb687e7367a98750779c6434ba794ac33963c2/pyvcloud/vcd/vdc.py#L345)):*
<details><summary><i>Previously, floriankammermann (Florian Kammermann) wrote…</i></summary><blockquote>

method naming is fixed in latest commit fixed in latest commit 
get the edge gateway list on the link is fixed in the latest commit

I had a problem to find the Link `rel="fullEdgeGateway"`, couldn't see it on my test vdc and I also couldn't find it in the docs https://pubs.vmware.com/vcd-820/index.jsp?topic=%2Fcom.vmware.vcloud.api.sp.doc_27_0%2FGUID-EECCA924-4DCE-490F-B352-5C212AD22411.html&resultof=%22%65%64%67%65%47%61%74%65%77%61%79%73%22%20%22%65%64%67%65%67%61%74%65%77%61%69%22%20%22%4c%69%6e%6b%22%20%22%6c%69%6e%6b%22%20
</blockquote></details>

`fullEdgeGateway` might not be present in all VDC resources, let's just focus on `edgeGateways` in this PR

---

*[pyvcloud/vcd/vdc.py, line 352 at r2](https://reviewable.io:443/reviews/vmware/pyvcloud/161#-L1tHMka31GbkjJ_ng-B:-L1tHMka31GbkjJ_ng-C:b-f6j56z) ([raw file](https://github.com/vmware/pyvcloud/blob/a6846b6b097053711d7991da545a175f74545abd/pyvcloud/vcd/vdc.py#L352)):*
> ```Python
>         if self.resource is None:
>             self.resource = self.client.get_resource(self.href)
>         links = get_links(
> ```

It will be better to use `Client.get_linked_resource`.
Also, once you get the list of records, you can just return an array with some attributes of the each record, (name, href). It should not retrieve the actual resource. See `org.list_vdcs` as a reference

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/pyvcloud/161#-:-L1tJiIZ45KUPjWmcEHk:bdugx8a)*

**Comment by pacogomez** [comment]
That should be sufficient for the `vcd gateway list` command. We should have an `info` command that will get the gateway resource from vCD and display it on stdout

---

Review status: 2 of 3 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/pyvcloud/161#-:-L1ywji0cTrHLfC1fA5J:bv2jew4)*

**Comment by pacogomez** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/pyvcloud/161#-:-L1yxCOjL0r77f2OtmDM:bnfp4nl)*

**Comment by floriankammermann** [comment]
Do I have something to do, to get it merged?

**Comment by pacogomez** [comment]
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/pyvcloud/161)*

**Comment by pacogomez** [comment]
Thanks for you contribution

**Pull request merged**